packages/table/src/Table.js

```diff
@@ -12,10 +12,34 @@ import { rowHeights } from "./constants";
 
 class Table extends React.Component {
   state = {
+    rowsMap: {},
     selectedRows: [],
     expandedRows: []
   };
 
+  componentDidMount() {
+    const { rowGroupKey } = this.props;
+    const hasSubRows = row =>
+      rowGroupKey && row[rowGroupKey] && row[rowGroupKey].length;
+
+    // iterate over all rows and sub-rows to create
+    // a hash table of rows indexed by keys in the
+    // format "<index>,<subIndex>"
+    const rowsMap = this.props.data.reduce((entries, row, index) => {
```

Comment:
What happens when rows are added to `props.data` after the component is mounted?

Reply:
would break selection checkboxes :P

Reply:
actually, I dont know, it would probably continue working ok, need to test it

Reply:
actually is the selectionData passed to selectionHeader that breaks, it wont pass a selected row if the data about that row was added afterwards, I will update the index on component update to fix this (and write tests)